### PR TITLE
*: partial deposits bugfixes

### PR DESCRIPTION
**Pull request by pinebit**
Several fixes for bugs identified in partial deposits implementation.
Plus, a few improvements to prevent using partial deposits for unsupported lock version.

category: bug
ticket: none

### CODE REVIEW

File: cluster/definition.go [hunk 1]
```diff
@@ -38,6 +38,13 @@ func WithVersion(version string) func(*Definition) {
 	}
 }
 
+// WithDKGAlgorithm returns an option to set a non-default DKG algorithm in a new definition.
+func WithDKGAlgorithm(algorithm string) func(*Definition) {
```
Comment (dB2510) [inline cluster/definition.go:42]: why is this needed? we only support one dkg algo atm

**Review by dB2510 — APPROVED** [review]

File: cluster/definition.go (hunk 1 above)
Comment (pinebit) [reply cluster/definition.go:42]: In createdkg.go it populates opts.. which we need to extend WithVersion.

Comment (dB2510) [reply cluster/definition.go:42]: but why do we need to populate the dkg algorithm?

Comment (pinebit) [reply cluster/definition.go:42]: I think this is to respect the given configuration. I believe the implementation assumes different algorithms passed from the config. If you think this is odd, then we can have a ticket to remove this field from the config entirely.
